src/Multivariate/ProductFun.jl

```diff
@@ -35,6 +45,15 @@ function ProductFun(cfs::AbstractMatrix{T},sp::AbstractProductSpace{Tuple{S,V},D
     end
 end
 
+## TODO: This Product Fun actually does not return a productfun, dirty but was easiest to implement. Probably an abstract type of ProductFuns
+# is needed in the future.
+function ProductFun(iter::TrivialTensorizer{d},cfs::Vector{T},blk::Block, sp::AbstractProductSpace{NTuple{d, S},DD}) where {S<:UnivariateSpace,T<:Number,d,DD}
```

Comment:
Currently multivariate functions are designed in a way to first build a tensor out of the coefficients and then first solve the first dimension followed by the second. `ProductFun` is designed for that, storing a vector of functions.
I first implemented it that way, but this needs too much memory for high dimensional cases and threw errors for me for dimensions larger about 80 and 10000 coefficients (~1 million coefficients are possible in the 1D case for me). Therefore, I never build this tensor and don't use the vector of functions `ProductFun` is using. I think the approach is not compatible with the current `ProductFun`.

I think the best way would actually be to change the order of how the coefficients are vectorized, see #86 , because this change in coefficients is computationally costly (functions `fromtensor` and `totensor` build the "real" tensor out of the current way the coefficients are stored. I introduced `totensoriterator` for evaluating d>2 dimensional functions to avoid building the full tensor).

However, this change could be breaking with current code using ApproxFun